libsolidity/formal/ModelCheckerSettings.cpp

```diff
@@ -82,36 +82,84 @@ map<TargetType, string> const ModelCheckerTargets::targetTypeToString{
 	{TargetType::OutOfBounds, "Out of bounds access"}
 };
 
-std::optional<ModelCheckerTargets> ModelCheckerTargets::fromString(string const& _targets)
+ModelCheckerTargets::ModelCheckerTargets(size_t nFlags, ...)
 {
-	set<TargetType> chosenTargets;
-	if (_targets == "default" || _targets == "all")
+	if (nFlags < 1)
+		return;
+
+	std::va_list args{};
+	va_start(args, nFlags);
+
+	for (size_t i = 0; i < nFlags; ++i)
 	{
-		bool all = _targets == "all";
-		for (auto&& v: targetStrings | ranges::views::values)
-		{
-			if (!all && (v == TargetType::Underflow || v == TargetType::Overflow))
-				continue;
-			chosenTargets.insert(v);
-		}
+		auto targetType = va_arg(args, VerificationTargetType);
+		setTargetType(targetType, true);
 	}
-	else
-		for (auto&& t: _targets | ranges::views::split(',') | ranges::to<vector<string>>())
-		{
-			if (!targetStrings.count(t))
-				return {};
-			chosenTargets.insert(targetStrings.at(t));
-		}
 
-	return ModelCheckerTargets{chosenTargets};
+	va_end(args);
 }
 
-bool ModelCheckerTargets::setFromString(string const& _target)
+bool ModelCheckerTargets::setTargetType(VerificationTargetType targetType, bool _value)
 {
-	if (!targetStrings.count(_target))
+	switch (targetType)
+	{
+	case VerificationTargetType::ConstantCondition:
+		constantCondition = _value;
+		return true;
+	case VerificationTargetType::Underflow:
+		underflow = _value;
+		return true;
+	case VerificationTargetType::Overflow:
+		overflow = _value;
+		return true;
+	case VerificationTargetType::UnderOverflow:
+		underOverflow = _value;
+		return true;
+	case VerificationTargetType::DivByZero:
+		divByZero = _value;
+		return true;
+	case VerificationTargetType::Balance:
+		balance = _value;
+		return true;
+	case VerificationTargetType::Assert:
+		assert = _value;
+		return true;
+	case VerificationTargetType::PopEmptyArray:
+		popEmptyArray = _value;
+		return true;
+	case VerificationTargetType::OutOfBounds:
+		outOfBounds = _value;
+		return true;
+	default:
 		return false;
-	targets.insert(targetStrings.at(_target));
-	return true;
+	}
+}
+
+bool ModelCheckerTargets::has(VerificationTargetType _type) const
+{
+	switch (_type)
+	{
+	case VerificationTargetType::ConstantCondition:
+		return constantCondition;
```

Comment:
Maybe we could add translation to/from an enum as a feature to the `FlagSet`? It's a bit unfortunate that with the new class this part actually got more repetitive.